src/Tests/Microsoft.NET.TestFramework/ExponentialRetry.cs

```diff
@@ -19,13 +19,13 @@ public static IEnumerable<TimeSpan> Intervals
                 while (true)
                 {
                     yield return TimeSpan.FromSeconds(seconds);
-                    seconds *= 2;
+                    seconds *= 10;
                 }
             }
         }
 
-        public static async Task ExecuteWithRetry(Func<Task<string>> action,
-            Func<string, bool> isSuccess,
+        public static async Task<T> ExecuteWithRetry<T>(Func<T> action,
+            Func<T, bool> successOrStopRetry,
```

Comment:
I'd simplify this name to something like `shouldStopRetry`, or invert it so it can be even simpler: `shouldRetry`.

Reply:
done